services/distribution/src/test/java/app/coronawarn/server/services/distribution/statistics/StatisticsJsonToProtobufTest.java

```diff
@@ -24,103 +31,150 @@
-          assertThat(keyFigureCard.getHeader()).isNotNull();
-          assertThat(keyFigureCard.getHeader().getUpdatedAt()).isPositive();
-        }
-    );
-  }
+    @Test
+    void testGetCardIdSequenceFromConfig() throws IOException {
+      Statistics stats = statisticsToProtobufMapping.constructProtobufStatistics();
 
-  @Test
-  void testKeyFigureCardBasedOnHeaderCardId() throws IOException {
-    Statistics stats = statisticsToProtobufMapping.constructProtobufStatistics();
+      assertThat(stats.getCardIdSequenceList().size()).isEqualTo(4);
+    }
 
-    KeyFigureCard infectionsCard = getKeyFigureCardForId(stats, 1);
-    KeyFigureCard incidenceCard = getKeyFigureCardForId(stats, 2);
-    KeyFigureCard keySubmissionsCard = getKeyFigureCardForId(stats, 3);
 
-    assertThat(infectionsCard.getKeyFiguresCount()).isEqualTo(3);
-    assertThat(incidenceCard.getKeyFiguresCount()).isEqualTo(1);
-    assertThat(keySubmissionsCard.getKeyFiguresCount()).isEqualTo(3);
-  }
+    @Test
+    void testKeyFigureCardContainsHeader() throws IOException {
+      Statistics stats = statisticsToProtobufMapping.constructProtobufStatistics();
+
+      assertThat(stats.getKeyFigureCardsCount()).isEqualTo(4);
+      stats.getKeyFigureCardsList().forEach(keyFigureCard -> {
+            assertThat(keyFigureCard.getHeader()).isNotNull();
+            assertThat(keyFigureCard.getHeader().getUpdatedAt()).isPositive();
+          }
+      );
+    }
 
-  @Test
-  void testEffectiveDateValidation() throws IOException {
-    StatisticsJsonValidator statisticsJsonValidator = new StatisticsJsonValidator();
-
-    String content = FileUtils.readFileToString(
-        new File("./src/test/resources/stats/statistic_data.json"), StandardCharsets.UTF_8);
-    List<StatisticsJsonStringObject> statsDTO = SerializationUtils.deserializeJson(content, typeFactory -> typeFactory
-        .constructCollectionType(List.class, StatisticsJsonStringObject.class));
-    statsDTO = new ArrayList<>(statisticsJsonValidator.validate(statsDTO));
-
-    assertThat(statisticsObjectContainsFields(statsDTO, "2020-12-04T11:13:22,588")).isTrue();
-    //The json object that has the effective_date set on null should not be anymore present after the validation
-    assertThat(statisticsObjectContainsFields(statsDTO, "2020-12-04T00:00:00,000")).isFalse();
-    //The json object that has the effective_date set on invalid format date should not be anymore present after the validation
-    assertThat(statisticsObjectContainsFields(statsDTO, "2020-12-05T00:01:00,000")).isFalse();
-  }
+    @Test
+    void testKeyFigureCardBasedOnHeaderCardId() throws IOException {
+      Statistics stats = statisticsToProtobufMapping.constructProtobufStatistics();
+
+      KeyFigureCard infectionsCard = getKeyFigureCardForId(stats, 1);
+      KeyFigureCard incidenceCard = getKeyFigureCardForId(stats, 2);
+      KeyFigureCard keySubmissionsCard = getKeyFigureCardForId(stats, 3);
 
```

Comment:
Maybe we can test here based on the header card Id the number of KeyFigureCard for the 4th card